source/extensions/filters/network/dubbo_proxy/serialization.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/singleton/const_singleton.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+
+/**
+ * Names of available serialization implementations.
+ */
+class SerializerNameValues {
+public:
+  // hessian serialization
+  const std::string HESSIAN = "hessian";
+  // json serialization
+  const std::string JSON = "json";
+};
+
+typedef ConstSingleton<SerializerNameValues> SerializerNames;
+
+/**
+ * RpcInvocation represent an rpc call
+ * See
+ * https://github.com/apache/incubator-dubbo/blob/master/dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcInvocation.java
+ */
+class RpcInvocation {
+public:
+  virtual ~RpcInvocation() {}
+  virtual const std::string& getMethodName() const PURE;
+  virtual const std::string& getServiceName() const PURE;
+  virtual const std::string& getServiceVersion() const PURE;
+};
+
+typedef std::unique_ptr<RpcInvocation> RpcInvocationPtr;
+
+/**
+ * RpcResult represent the result of an rpc call
+ * See
+ * https://github.com/apache/incubator-dubbo/blob/master/dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/RpcResult.java
+ */
+class RpcResult {
+public:
+  virtual ~RpcResult() {}
+  virtual bool hasException() const PURE;
+};
+
+typedef std::unique_ptr<RpcResult> RpcResultPtr;
+
+class SerializationCallbacks {
+public:
+  virtual ~SerializationCallbacks() {}
+  virtual void onRpcInvocation(RpcInvocationPtr&& invo) PURE;
+  virtual void onRpcResult(RpcResultPtr&& res) PURE;
+};
+
+class Serialization {
```

Comment:
(throughout this PR) this should be named `Deserializer`?

Reply:
fixed